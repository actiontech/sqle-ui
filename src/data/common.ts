export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}