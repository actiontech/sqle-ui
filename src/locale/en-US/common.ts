// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: 'username',
  password: 'password',

  unknownError: 'Unknown Error',

  ok: 'Ok',
  cancel: 'Cancel',
  submit: 'Submit',
  close: 'Close',
  edit: 'Edit',
  delete: 'Delete',
  reset: 'Reset',
  search: 'Search',
  retry: 'Retry',
  back: 'Back',
  upload: 'Upload',
  resetAndClose: 'Reset Form And Close',
  operateSuccess: 'Operate successfully',
  prevStep: 'Prev',
  nextStep: 'Next',

  expansion: 'Expand',
  collapse: 'Collapse',

  theme: {
    light: 'Light Theme',
    dark: 'Dark Theme',
  },

  logout: 'Logout',
  account: 'Account',

  nav: {
    title: 'SQLe',
  },

  request: {
    noticeFailTitle: 'Request Failed',
  },

  form: {
    placeholder: {
      input: 'Please input {{name}}',
      select: 'Please select {{name}}',
      searchInput: 'Input the {{name}} to search',
      searchSelect: 'Select the {{name}} to search',
    },
    rule: {
      require: 'Please input {{name}}',
      passwordNotMatch: 'The two passwords that you entered do not match!',
      email: 'The input is not valid E-mail!',
    },
  },
};
