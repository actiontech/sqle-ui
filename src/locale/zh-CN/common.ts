// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: '用户名',
  password: '密码',

  unknownError: '未知错误...',
  unknownStatus: '未知状态...',

  ok: '确认',
  cancel: '取消',
  submit: '提交',
  close: '关闭',
  edit: '编辑',
  modify: '修改',
  delete: '删除',
  reset: '重置',
  resetAll: '重置所有内容',
  search: '搜索',
  retry: '重试',
  back: '返回',
  more: '更多',
  upload: '选择文件上传',
  resetAndClose: '重置表单并关闭',
  operateSuccess: '操作成功',
  operate: '操作',

  prevStep: '上一步',
  nextStep: '下一步',

  expansion: '展开',
  collapse: '收起',

  showAll: '查看所有',
  showDetail: '查看详情',

  in: '在',
  on: '的',
  and: '且',
  at: '中',
  preview: '预览',

  theme: {
    light: '明亮模式',
    dark: '暗黑模式',
  },

  logout: '退出登录',
  account: '个人中心',

  nav: {
    title: 'SQLE',
  },

  request: {
    noticeFailTitle: '请求错误',
  },

  time: {
    hour: '小时',
    year: '年',
    month: '月',
    day: '天',
    no: '日',
    week: '星期',
    minute: '分钟',
    per: '每',
  },

  form: {
    placeholder: {
      input: '请输入{{name}}',
      select: '请选择{{name}}',
      searchInput: '请输入要搜索的 {{name}}',
      searchSelect: '请选择要搜索的 {{name}}',
    },
    rule: {
      require: '必须填写{{name}}',
      selectFile: '必须选择一个文件',
      passwordNotMatch: '两次密码输入不一致',
      email: '请输入有效的邮箱地址',
      startWithLetter: '必须要以字母开头',
      onlyLetterAndNumber: '只能包含字母、数字、中划线和下划线',
    },
  },

  cron: {
    mode: {
      select: '可视化选择',
      manual: '手工填写',
    },
    errorMessage: {
      invalid: '无效的cron表达式',
      mustBeString: 'cron表达式必须是一个字符串',
      mustBeArray: '变更的值必须是一个数组',
      lenMustBeFive: 'cron表达式必须只包含（分钟 小时 日期 月份 星期）5个元素',
      onlyHaveValidChar: 'cron表达式只能包含数字中划线(-),斜线(/),和逗号(,)',
      limit:
        '您的表达式中包含不合法的数值范围， minute(0-59), hour(0-23), day(1,31), month(1-12), week(0-6)',
    },
  },
};
