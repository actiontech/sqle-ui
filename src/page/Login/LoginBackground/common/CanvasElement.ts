import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
