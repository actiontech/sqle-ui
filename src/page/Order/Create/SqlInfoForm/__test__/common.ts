import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
