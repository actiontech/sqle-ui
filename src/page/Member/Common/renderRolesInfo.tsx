import { Space, Typography } from 'antd';
import { IBindRoleReqV1 } from '../../../api/common';

const renderRolesInfo = (roles: IBindRoleReqV1[], ellipsis: boolean) => {
  const genContent = (roleNames?: string[], instanceName?: string) =>
    `${instanceName ?? ''}: [ ${roleNames?.toString() ?? ''} ]`;
  return (
    <Space direction="vertical">
      {roles.map((v) => {
        return ellipsis ? (
          <Typography.Text ellipsis={ellipsis} key={v.instance_name}>
            {genContent(v.role_names, v.instance_name)}
          </Typography.Text>
        ) : (
          <div key={v.instance_name}>
            {genContent(v.role_names, v.instance_name)}
          </div>
        );
      })}
    </Space>
  );
};

export default renderRolesInfo;
