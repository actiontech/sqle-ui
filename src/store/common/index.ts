import { PayloadAction } from '@reduxjs/toolkit';
import { ModalStatus } from '../../types/common.type';

export const commonModalReducer = <
  T extends { modalStatus: ModalStatus }
>() => ({
  initModalStatus(
    state: T,
    { payload: { modalStatus } }: PayloadAction<{ modalStatus: ModalStatus }>
  ) {
    state.modalStatus = modalStatus;
  },
  updateModalStatus(
    state: T,
    {
      payload: { modalName, status },
    }: PayloadAction<{ modalName: string; status: boolean }>
  ) {
    if (Object.prototype.hasOwnProperty.call(state.modalStatus, modalName)) {
      state.modalStatus[modalName] = status;
    }
  },
});
