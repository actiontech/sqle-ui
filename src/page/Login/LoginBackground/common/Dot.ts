import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
