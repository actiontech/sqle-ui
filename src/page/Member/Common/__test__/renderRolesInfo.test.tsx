import { IBindRoleReqV1 } from '../../../../api/common';
import renderRolesInfo from '../renderRolesInfo';

describe('test renderRolesInfo', () => {
  const roles: IBindRoleReqV1[] = [
    {
      instance_name: 'db1',
      role_names: ['test1', 'test2', 'test3'],
    },
    {
      instance_name: 'db2',
      role_names: ['test3'],
    },
    {
      instance_name: 'db3',
      role_names: ['test3', 'test4', 'test5', 'test6', 'test7'],
    },
  ];
  test('should match snapshot', () => {
    expect(renderRolesInfo(roles, false)).toMatchSnapshot();
    expect(renderRolesInfo(roles, true)).toMatchSnapshot();
  });
});
