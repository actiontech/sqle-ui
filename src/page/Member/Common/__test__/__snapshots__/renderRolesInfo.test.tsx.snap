// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renderRolesInfo should match snapshot 1`] = `
<Space
  direction="vertical"
>
  <div>
    db1: [ test1,test2,test3 ]
  </div>
  <div>
    db2: [ test3 ]
  </div>
  <div>
    db3: [ test3,test4,test5,test6,test7 ]
  </div>
</Space>
`;

exports[`test renderRolesInfo should match snapshot 2`] = `
<Space
  direction="vertical"
>
  <ForwardRef(Text)
    ellipsis={true}
  >
    db1: [ test1,test2,test3 ]
  </ForwardRef(Text)>
  <ForwardRef(Text)
    ellipsis={true}
  >
    db2: [ test3 ]
  </ForwardRef(Text)>
  <ForwardRef(Text)
    ellipsis={true}
  >
    db3: [ test3,test4,test5,test6,test7 ]
  </ForwardRef(Text)>
</Space>
`;
